yarn-project/stdlib/src/logs/log_with_tx_data.ts

```diff
@@ -1,35 +1,51 @@
-import { MAX_NOTE_HASHES_PER_TX, PUBLIC_LOG_SIZE_IN_FIELDS } from '@aztec/constants';
+import { MAX_NOTE_HASHES_PER_TX, PRIVATE_LOG_SIZE_IN_FIELDS, PUBLIC_LOG_SIZE_IN_FIELDS } from '@aztec/constants';
 import { Fr } from '@aztec/foundation/fields';
 import { TxHash } from '@aztec/stdlib/tx';
 
-// TypeScript representation of the Noir aztec::oracle::message_discovery::PublicLogWithTxData struct. This is used as a
-// response for PXE's custom getPublicLogByTagFromContract oracle.
-export class PublicLogWithTxData {
+// TypeScript representation of the Noir aztec::oracle::message_discovery::LogWithTxData struct.
+class LogWithTxData<N extends number> {
   constructor(
     // The emitted fields of a log.
-    // For public logs, the contract address is prepended to the content.
     public logContent: Fr[],
     public txHash: TxHash,
     public uniqueNoteHashesInTx: Fr[],
     public firstNullifierInTx: Fr,
+    private maxLogContentLength: N,
   ) {}
 
   toNoirSerialization(): (Fr | Fr[])[] {
     return [
-      // The log fields length is PUBLIC_LOG_SIZE_IN_FIELDS. + 1 because the contract address is prepended to the content.
-      // This is only used for public logs currently, so the maxLength is PUBLIC_LOG_SIZE_IN_FIELDS + 1.
-      ...toBoundedVecSerialization(this.logContent, PUBLIC_LOG_SIZE_IN_FIELDS + 1),
+      ...toBoundedVecSerialization(this.logContent, this.maxLogContentLength),
       this.txHash.hash,
       ...toBoundedVecSerialization(this.uniqueNoteHashesInTx, MAX_NOTE_HASHES_PER_TX),
       this.firstNullifierInTx,
     ];
   }
+}
+
+// The total size of a public log in fields, including the contract address.
+const TOTAL_PUBLIC_LOG_SIZE_IN_FIELDS = PUBLIC_LOG_SIZE_IN_FIELDS + 1;
```

Comment:
I am very not sure if this ^ is correct. I see that in `constants.nr` we have:

```
pub global PUBLIC_LOG_SIZE_IN_FIELDS: u32 = 13;
pub global PUBLIC_LOG_LENGTH: u32 = PUBLIC_LOG_SIZE_IN_FIELDS
    + 1 /* length */
    + 1 /* contract_address */;
```

This ^ would indicate that the TOTAL_PUBLIC_LOG_SIZE_IN_FIELDS should be 15 but here I have 14. Does the reviewer know if the length gets trimmed somewhere? Or should this really be 15?

Reply:
No longer relevant